src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -25,6 +25,9 @@ namespace Microsoft.Build.Logging.TerminalLogger;
 /// </remarks>
 internal sealed class TerminalLogger : INodeLogger
 {
+    private const string FilePathPattern = " -> ";
+    private readonly string[] _immediateMessageMarkers = new[] { "[CredentialProvider]", "--interactive" };
```

Comment:
nit: .NET 8 provides a SearchValues<T> structure that, when used in `IndexOfAny` gives massive perf speedups. can you see if that works, since we'll be doing this on every message?

Reply:
we compile for .net framework too, is it ok to use `IndexOfAny`for .net only?

Reply:
SearchValues only applies to sets of individual characters, as far as I can tell.